core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -51,8 +54,39 @@ private[spark] class TaskDescription(
     val index: Int,    // Index within this task's TaskSet
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
-    val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    val properties: Properties) extends Logging {
+
+  def this(
+      taskId: Long,
+      attemptNumber: Int,
+      executorId: String,
+      name: String,
+      index: Int, // Index within this task's TaskSet
+      addedFiles: Map[String, Long],
+      addedJars: Map[String, Long],
+      properties: Properties,
+      task: Task[_]) {
+      this(taskId, attemptNumber, executorId, name, index,
+        addedFiles, addedJars, properties)
+      task_ = task
```

Comment:
I found the current code very hard to follow, and it sounds like @squito had a similar experience.  It's confusing to have two constructors that are very similar: it makes it hard for future users of this code to know which constructor they should be using, and hard for readers of the code to understand why one constructor would be called.  I'm open to other suggestions for how to simplify the code than the one I suggested -- but in it's current form I find it to be too confusing.